commands.go

```diff
@@ -152,6 +152,13 @@ type Cmdable interface {
 	SRem(key string, members ...interface{}) *IntCmd
 	SUnion(keys ...string) *StringSliceCmd
 	SUnionStore(destination string, keys ...string) *IntCmd
+	XAdd(key string, id string, els ...interface{}) *StringCmd
+	XAddMaxLen(key string, id string, maxLen int64, approximate bool, els ...interface{}) *StringCmd
+	XLen(key string) *IntCmd
+	XRange(key string, start, stop string) *XMessageSliceCmd
+	XRangeN(key string, start, stop string, count int64) *XMessageSliceCmd
+	XRead(block int64, streams []XStreamCheckpoint) *XStreamSliceCmd
```

Comment:
Can we change to `block time.Duration`? That will make API cleaner.

Reply:
👍
What are your thoughts about `BLOCK 0` (block forever until new message(s) appear in the stream)? Currently this will wait as long as the client's `ReadTimeout` allows, then give an `i/o timeout` error. 

Should `XRead` be immune to these since it's intended functionality?

Reply:
Yeah, library already allows you to set read timeout https://github.com/go-redis/redis/blob/master/commands.go#L947. Zero timeout is also supported.